zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java

```diff
@@ -235,7 +235,11 @@ private static ServletContextHandler setupRestApiContextHandler(ZeppelinConfigur
     cxfContext.addFilter(new FilterHolder(CorsFilter.class), "/*",
         EnumSet.allOf(DispatcherType.class));
 
-    cxfContext.addFilter(org.apache.shiro.web.servlet.ShiroFilter.class, "/*",
+    FilterHolder shiroFilter =  new FilterHolder(ShiroFilter.class);
+    shiroFilter.setInitParameter("shiroConfigLocations", "file:../"
+        + conf.getConfDir() + "/shiro.ini");
```

Comment:
`conf.getConfDir()` can be either relative path or absolute path. So in case of absolute path, it might not work as expected.

How about add one method in ZeppelinConfiguration to get path of shiro.ini, such as getShiroDir()? similar to https://github.com/apache/incubator-zeppelin/blob/master/zeppelin-zengine/src/main/java/org/apache/zeppelin/conf/ZeppelinConfiguration.java#L345

And then `new File(conf.getShiroDir()).toURI().toString()` will get uri string that work for both relative and absolute path.